app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionFragment.kt

```diff
@@ -96,14 +95,12 @@ class DataCollectionFragment : AbstractFragment(), BackPressListener {
       }
     )
 
-    updateUI(
-      UiState.TaskListAvailable(
-        viewModel.tasks,
-        viewModel.getTaskPosition(viewModel.getCurrentTaskId()),
-      )
-    )
-
-    lifecycleScope.launch { viewModel.uiState.filterNotNull().collect { updateUI(it) } }
+    // Collect UI state safely across the Fragment view lifecycle.
+    viewLifecycleOwner.lifecycleScope.launch {
+      viewLifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+        viewModel.uiState.collect { ui -> updateUI(ui) }
+      }
+    }
```

Comment:
Why this is change necessary?

```
viewLifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED)
```

Reply:
We need repeatOnLifecycle so that our flow collection is bound to the Fragment’s view lifecycle state.